subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/TypeMetadataWalker.java

```diff
@@ -21,20 +21,80 @@
 import java.lang.annotation.Annotation;
 import java.util.function.Supplier;
 
+/***
+ * A generalized Type Metadata Walker for Gradle types or instances.
+ *
+ * A walker visits all properties of a type or an instance.
```

Comment:
🤔 We have a lot of detail now in the Javadocs on the methods, some of which are trivial, and as such don't add much. I'd probably remove some of that — e.g. the `@param` docs seem mostly redundant.

I would instead add information about:

- the relationship between the walker and the metadata store
- the definition of "property" in this context
- how this relates to annotations
- the definition of a "nested" property

I think it's fine to add all of this info in the general Javadoc of the interface, and leave the methods largely undocumented.

Reply:
I did some changes to docs. Let me know what you think.